telemetry/util/heka_message.py

```diff
@@ -0,0 +1,205 @@
+    # Read 1 byte record separator (and keep reading until we get one)
+    total_bytes = 0
+    skipped, eof = read_until_next(input_stream, 0x1e)
+    total_bytes += skipped
+    if eof:
+        return None, total_bytes
+    else:
+        # we've read one separator (plus anything we skipped)
+        total_bytes += 1
+
+    if skipped > 0:
+        if strict:
+            raise ValueError("Unexpected character(s) at the start of record")
+        if verbose:
+            print "Skipped", skipped, "bytes to find a valid separator"
+
+    raw_record = struct.pack("<B", 0x1e)
+
+    # Read the header length
+    header_length_raw = input_stream.read(1)
+    if header_length_raw == '':
+        return None, total_bytes
+
+    total_bytes += 1
+    raw_record += header_length_raw
+
+    # The "<" is to force it to read as Little-endian to match the way it's
+    # written. This is the "native" way in linux too, but might as well make
+    # sure we read it back the same way.
+    (header_length,) = struct.unpack('<B', header_length_raw)
+
+    header_raw = input_stream.read(header_length)
+    if header_raw == '':
+        return None, total_bytes
+    total_bytes += header_length
+    raw_record += header_raw
+
+    header = message_pb2.Header()
+    header.ParseFromString(header_raw)
+    unit_separator = input_stream.read(1)
+    total_bytes += 1
+    if ord(unit_separator[0]) != 0x1f:
+        error_msg = "Unexpected unit separator character in record #{} " \
+                "at offset {}: {}".format(record_count, total_bytes,
+                ord(unit_separator[0]))
+        if strict:
+            raise ValueError(error_msg)
+        return UnpackedRecord(raw_record, header, error=error_msg), total_bytes
+    raw_record += unit_separator
+
+    #print "message length:", header.message_length
+    message_raw = input_stream.read(header.message_length)
+
+    total_bytes += header.message_length
+    raw_record += message_raw
+
+    message = None
+    if not raw:
+        message = message_pb2.Message()
+        parsed_ok = False
+        if try_snappy:
+            try:
+                message.ParseFromString(snappy.decompress(message_raw))
+                parsed_ok = True
+            except:
+                # Wasn't snappy-compressed
+                pass
+        if not parsed_ok:
+            # Either we didn't want to attempt snappy, or the
+            # data was not snappy-encoded (or it was just bad).
+            message.ParseFromString(message_raw)
+
+    return UnpackedRecord(raw_record, header, message), total_bytes
+
+
+def unpack_file(filename, **kwargs):
+    fin = None
+    if filename.endswith(".gz"):
+        fin = gzip.open(filename, "rb")
+    else:
+        fin = open(filename, "rb")
+    return unpack(fin, **kwargs)
+
+
+def unpack_string(string, **kwargs):
+    return unpack(StringIO(string), **kwargs)
+
+
+def unpack(fin, raw=False, verbose=False, strict=False, backtrack=False, try_snappy=False):
```

Comment:
let's set the default for `try_snappy` to `True` so we don't have to revisit this.